docs/installation.md

```diff
@@ -0,0 +1,638 @@
+   - Follow `Clients > transmart-client > Roles` (tab)
+   - Create client role `ROLE_ADMIN`:
+
+     > ![Administrator role](images/Administrator%20role.png)
+
+4. Add users\
+   Keycloak supports [identity brokering](https://www.keycloak.org/docs/latest/server_admin/index.html#_identity_broker), or [user federation](https://www.keycloak.org/docs/latest/server_admin/index.html#_user-storage-federation) for integration with existing organisational identity providers.\
+   To add users manually:
+   - Go to `Users`
+   - Click `Add user`, enter a username and click `Save`.
+   - Select the `Credentials` tab, enter a password and click `Reset Password`.
+   - If the user should have administrator access:
+     - Go to `Role Mappings` tab.
+     - Then select `transmart-client` in the `Client Roles` dropdown and assure that `ROLE_ADMIN` is in the list of assigned roles.
+
+5. Add a system user for asynchronous job handling by the Glowing Bear backend Transmart Packer.
+   - Go to `Users`
+   - Click `Add user`, enter username `system` and click `Save`.
+   - Select the `Credentials` tab, enter a strong password and click `Reset Password`.
+   - Go to `Role Mappings` tab.
+     - Ensure that the `offline_access` realm role is assigned.
+     - Select `realm-management` in the `Client Roles` dropdown and assure the `view-users` and `impersonation` roles are assigned.
+
+     > ![System user roles](images/System%20user%20roles.png)
+
+#### Offline token
+
+Request an offline token for the `system` user, to be used by Glowing Bear backend and Transmart Packer. 
+
+```bash
+KEYCLOAK_CLIENT_ID=transmart-client
+SYSTEM_USERNAME=system
+SYSTEM_PASSWORD=choose-a-strong-system-password # CHANGE ME
+KEYCLOAK_SERVER_URL=https://keycloak.example.com
+KEYCLOAK_REALM=example
+
+curl -f --no-progress-meter \
+  -d "client_id=${KEYCLOAK_CLIENT_ID}" \
+  -d "username=${SYSTEM_USERNAME}" \
+  -d "password=${SYSTEM_PASSWORD}" \
+  -d 'grant_type=password' \
+  -d 'scope=offline_access' \
+  "${KEYCLOAK_SERVER_URL}/realms/${KEYCLOAK_REALM}/protocol/openid-connect/token' | jq -r '.refresh_token'
+```
+
+The value of the `refresh_token` field in the response is the offline token.
+
+#### Availability check for monitoring
+
+Availability of the Keycloak application can be tested using the following public endpoints:
+- The realm specific endpoint `/realms/example/.well-known/openid-configuration` should return status `200`.
+- The health endpoint `/health` should return a JSON message with value `UP` in the `status` field:
+  ```json
+  {
+    "status": "UP",
+    "checks": [
+      {
+        "name": "Keycloak database connections health check",
+        "status": "UP"
+      }
+    ]
+  }
+  ```
+
+### TranSMART API server and database
+
+TranSMART API server requires:
+- JDK 8
+- a PostgreSQL database server
+
+Installation steps for TranSMART API server:
+
+1. Install the `pg_bitcount` extension, e.g., for PostgreSQL 11:
+   ```shell
+   POSTGRESQL_VERSION=12
```

Comment:
Not a problem, but seems a bit inconsistent that line above it mentions `e.g., for PostgreSQL 11` and here is `12`